Security/Firewall/GetJWTListener.php

```diff
@@ -98,17 +99,27 @@ public function handle(GetResponseEvent $event)
     {
         $request = $event->getRequest();
 
-        if ($this->options['post_only'] && !$request->isMethod('POST')) {
-            $event->setResponse(new JsonResponse('invalid method', 405));
-            return;
+        $is_json_request = $request->headers->get('content_type') === 'application/json';
+
+        if ($is_json_request) {
+            if (empty($request->getContent())) {
+                throw new \Exception('Error: Request parameters not found.');
```

Comment:
Throwing the base `Exception` class is not a good idea. There are more specific classes that we can use. "Error: " in the message is extraneous.